submission/proxy_test.go

```diff
@@ -125,15 +142,40 @@ func TestProxyInitState(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
 	defer cancel()
 
+Init:
 	for {
 		select {
 		case <-ctx.Done():
 			t.Fatalf("p.Run() expected to send init signal, got none")
-		case b := <-p.Init:
+		case b, ok := <-p.Init:
+			if !ok {
+				t.Fatalf("p.Run() expected to send 'true' init signal via Init channel, but channel is closed")
+			}
 			if b != true {
 				t.Fatalf("p.Run() expected to send 'true' init signal, got false")
 			}
+			break Init
+		case e := <-p.Errors:
+			t.Log(e)
+		}
+	}
+
+	sampleLogListUpdate := strings.Replace(testdata.SampleLogList, "ct.googleapis.com/racketeer/", "ct.googleapis.com/racketeer/v2/", 1)
+	err = updateFile(f, sampleLogListUpdate)
+	if err != nil {
+		t.Fatalf("unable to update Log-list data file: %q", err)
+	}
+	ctx, cancel = context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	p.llWatcher.RefreshLogListAndNotify(ctx)
```

Comment:
Rather than needing this new method, could you not instead change the argument to `p.Run()` from `time.Hour` to `time.Millisecond` in order to trigger a log list update?

Reply:
Done. Yet, leaving the helper func as it slightly simplifies the code-layout.